### PR TITLE
Modify mission control behavior tests

### PR DESCRIPTION
**Pull request by patapufate**
# Description
This PR:
- implements the changes made in #110 -> enable mask calculation.
- Check if the behavior had a failure state before the mission sends "complete" state.
- Add test to mission_control test_mission_control_behaviors.test

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [x] Unit tests are passing
- [x] Linter tests are passing (roslint pep8)

# How To Test

```sh
rostest mission_control test_mission_control_behaviors.test
```

### CODE REVIEW

File: mission_control/test/test_attitude_servo_behavior.py [hunk 1]
```diff
@@ -115,9 +118,12 @@ def attitude_servo_goals_are_set():
         self.simulated_auv_interface_data_pub.publish(msg)
 
         def success_mission_status_is_reported():
-            return self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE
+            if self.mission.execute_mission_state == ReportExecuteMissionState.ABORTING:
+                self.mission_has_failed = True
+            return (self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE and
+                    not self.mission_has_failed)
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:124]: @patapufate +1 to check for state transitions, but this check is flaky. It doesn't check that the sequence of states was correct, and it races with mission control to detect the temporary ABORTING state. 

I'd suggest we only check to final states OR that we do checks on a list of all reported states so far.

**Comment by hidmic** [comment]
To summarize my thoughts, I think we should revisit all these tests and make sure they don't flake. I'd also suggest using some helpful abstractions like futures to boost code reduce.

File: mission_control/test/test_attitude_servo_behavior.py (hunk 1 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:124]: @hidmic I have implemented only in altitude behavior test before move to all. PTAL a84d649

File: mission_control/test/test_attitude_servo_behavior.py [hunk 2]
```diff
@@ -115,10 +117,10 @@ def attitude_servo_goals_are_set():
         self.simulated_auv_interface_data_pub.publish(msg)
 
         def success_mission_status_is_reported():
-            return self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE
+            return (not ReportExecuteMissionState.ABORTING in self.mission.execute_mission_state and
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:120]: @patapufate same as above.

File: mission_control/test/test_depth_heading_behavior.py [hunk 3]
```diff
@@ -109,9 +112,11 @@ def depth_heading_goals_are_set():
         self.simulated_auv_interface_data_pub.publish(auv_interface_data)
 
         def success_mission_status_is_reported():
-            return self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE
+            return (not ReportExecuteMissionState.ABORTING in self.mission.execute_mission_state and
```
Comment (hidmic) [inline mission_control/test/test_depth_heading_behavior.py:115]: @patapufate same as above.

File: mission_control/test/test_fixed_rudder_behavior.py [hunk 4]
```diff
@@ -80,20 +80,33 @@ def test_mission_with_fixed_rudder_behavior(self):
...
+                    (speed_knots is None or self.fixed_rudder_goal.speed_knots == float(speed_knots)) and
+                    self.fixed_rudder_goal.ena_mask == enable_mask)
+
         self.assertTrue(wait_for(fixed_rudder_goals_are_set),
                         msg='Mission control must publish goals')
 
         # Wait for the mission to be complete
         def success_mission_status_is_reported():
-            return self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE
+            return (not ReportExecuteMissionState.ABORTING in self.mission.execute_mission_state and
```
Comment (hidmic) [inline mission_control/test/test_fixed_rudder_behavior.py:106]: @patapufate same as above.

File: mission_control/test/test_if_mission_control_aborts_when_behavior_returns_time_out_failure.py [hunk 5]
```diff
@@ -81,9 +81,9 @@ def test_mission_control_aborts_if_health_monitor_reports_fault(self):
 
         # Wait for the mission returns time_out Failure
         def aborting_mission_status_is_reported():
-            return self.mission.execute_mission_state == ReportExecuteMissionState.ABORTING
+            return ReportExecuteMissionState.ABORTING in self.mission.execute_mission_state
         self.assertTrue(wait_for(aborting_mission_status_is_reported),
-                        msg='Mission control must report ABORTING')
+                        msg='Mission control must reports ABORTING')
```
Comment (hidmic) [inline mission_control/test/test_if_mission_control_aborts_when_behavior_returns_time_out_failure.py:86]: @patapufate nit: 
```suggestion
                        msg='Mission control must report ABORTING')
```

**Review by hidmic — APPROVED** [review]
LGTM but for a few minor comments!

File: mission_control/test/test_altitude_heading_behavior.py [hunk 6]
```diff
@@ -113,9 +113,10 @@ def altitude_heading_goals_are_set():
         self.simulated_auv_interface_data_pub.publish(auv_interface_data)
 
         def success_mission_status_is_reported():
-            return self.mission.execute_mission_state == ReportExecuteMissionState.COMPLETE
+            return (not ReportExecuteMissionState.ABORTING in self.mission.execute_mission_state and
```
[the remark this answers is not in the document]
Comment (patapufate) [reply mission_control/test/test_altitude_heading_behavior.py:116]: Fixed

File: mission_control/test/test_attitude_servo_behavior.py (hunk 2 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:120]: Fixed

File: mission_control/test/test_depth_heading_behavior.py (hunk 3 above)
Comment (patapufate) [reply mission_control/test/test_depth_heading_behavior.py:115]: Fixed

File: mission_control/test/test_fixed_rudder_behavior.py (hunk 4 above)
Comment (patapufate) [reply mission_control/test/test_fixed_rudder_behavior.py:106]: fixed

File: mission_control/test/test_if_mission_control_aborts_when_behavior_returns_time_out_failure.py (hunk 5 above)
Comment (patapufate) [reply mission_control/test/test_if_mission_control_aborts_when_behavior_returns_time_out_failure.py:86]: Fixed

**Pull request merged**